### PR TITLE
[reboot-history] Add reboot history to state db

### PR DESCRIPTION
**Pull request by sujinmkang**
**- Why I did it**
Add reboot history to State db so that can be used telemetry service
**- How I did it**
Split the process-reboot-cause service to determine-reboot-cause and process-reboot-cause 
determine-reboot-cause to determine the reboot cause
process-reboot-cause to parse the reboot cause files and put the reboot history to state db
Moved to sonic-host-service* packages
**- How to verify it**
Performed unit test and tested on DUT
**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by sujinmkang** [comment]
retest vsimage please

File: src/sonic-host-services/scripts/process-reboot-cause [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+
+# ============================= Functions =============================
+def read_reboot_cause_files_and_save_state_db():
+    # Connect State DB
+    state_db = swsssdk.SonicV2Connector(host=REDIS_HOSTIP)
+    state_db.connect(state_db.STATE_DB)
+
+    # Sort the previous reboot cause files by creation time
+    REBOOT_FILE_LIST = [os.path.join(REBOOT_CAUSE_HISTORY_DIR,i) for i in os.listdir(REBOOT_CAUSE_HISTORY_DIR)]
```
Comment (jleveque) [inline src/sonic-host-services/scripts/process-reboot-cause:46]: Add space after comma

File: src/sonic-host-services/scripts/process-reboot-cause [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    TIME_SORTED_FULL_REBOOT_FILE_LIST = sorted(REBOOT_FILE_LIST, key=os.path.getmtime, reverse=True)
+
+    data = []
+    # Read each sorted previous reboot cause file and update the state db with previous reboot cause information
+    for i in range(min(10,len(TIME_SORTED_FULL_REBOOT_FILE_LIST))):
```
Comment (jleveque) [inline src/sonic-host-services/scripts/process-reboot-cause:51]: Add space after comma

File: src/sonic-host-services/setup.py [hunk 3]
```diff
@@ -1,5 +1,11 @@
 from setuptools import setup
 
+dependencies = [
+    'natsort',
+    'sonic_py_common',
+    'swsssdk>=2.0.1',
+]
```
Comment (jleveque) [inline src/sonic-host-services/setup.py:7]: `dependencies` is not being used. Please remove this and add these dependencies to the `install_requires` list below instead. Also, it doesn't look like `natsort` is being used and the other two dependencies are already there.

**Review by jleveque — CHANGES_REQUESTED** [review]

**Review by jleveque — APPROVED** [review]

**Review by jleveque — CHANGES_REQUESTED** [review]
Please fix LGTM alert.

**Review by jleveque — APPROVED** [review]

**Comment by sujinmkang** [comment]
retest vsimage please

**Pull request merged**